File: TypeScript/codewars/Typescript/Find-the-vowels.ts
export function vowelIndices(word: string): number[] {
	const result: number[] = [];
	const vowels = new Set(["a", "e", "i", "o", "u", "y"]);

	for (let index = 0; index < word.length; index++) {
		const char = word[index].toLowerCase();
		if (vowels.has(char)) {
			result.push(index + 1);
		}
	}

	return result;
}

File: Csharp/CodeWars/src/Resolutions/FindTheVowels.cs
namespace CodeWars.Resolutions;

public class FindTheVowels
{
    private static readonly string vowels = "aeiouy";
    public static int[] VowelIndices(string word) =>
        [.. word
        .ToLower()
        .Select((value, index) => new { value, index = index + 1 })
        .Where(x => vowels.Contains(x.value))
        .Select(x => x.index)];
}


